### PR TITLE
[e2e] Print IP config IDs if subnet deletion fails

### PR DESCRIPTION
**Pull request by lzhecheng**
#### What type of PR is this?
/kind testing

#### What this PR does / why we need it:
[e2e] Print IP config IDs if subnet deletion fails
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: tests/e2e/utils/network_utils.go [hunk 1]
```diff
@@ -127,14 +127,21 @@ func (azureTestClient *AzureTestClient) DeleteSubnet(vnetName string, subnetName
 			return false, nil
```
Comment (feiskyer) [inline tests/e2e/utils/network_utils.go:127]: Could you get the DELETE operation failure and log the delete error as well?

Comment (lzhecheng) [reply tests/e2e/utils/network_utils.go:128]: Updated. PTAL.

Comment (lzhecheng) [reply tests/e2e/utils/network_utils.go:128]: Updated. Used `%w`

Comment (nilo19) [reply tests/e2e/utils/network_utils.go:128]: %w is used for wrapping an error into another one, not for logging. I think we can use %s for err.Error() instead.

Comment (lzhecheng) [reply tests/e2e/utils/network_utils.go:128]: Makes sense. Used `%q` and `err.Error()` now.

**Pull request merged**